st2api/st2api/controllers/resource.py

```diff
@@ -118,19 +118,22 @@ def __init__(self):
 
         self.get_one_db_method = self._get_by_name_or_id
 
-    def get_all(self, **kwargs):
-        return self._get_all(**kwargs)
+    def get_all(self, sort=None, offset=0, limit=None, **raw_filters):
+        return self._get_all(sort=sort,
+                             offset=offset,
+                             limit=limit,
+                             raw_filters=raw_filters)
 
     def get_one(self, id):
         return self._get_one_by_id(id=id)
 
     def _get_all(self, exclude_fields=None, sort=None, offset=0, limit=None, query_options=None,
-                 from_model_kwargs=None, **raw_filters):
+                 from_model_kwargs=None, raw_filters=None):
         """
         :param exclude_fields: A list of object fields to exclude.
         :type exclude_fields: ``list``
         """
-        raw_filters = copy.deepcopy(raw_filters)
+        raw_filters = copy.deepcopy(raw_filters) or {}
```

Comment:
I guess `copy.deepcopy` also works on `None` type so it's probably fine and we don't need additional if.

And I would imagine it also doesn't cause any negative performance overhead since `None` is a built-in type and `copy.deepcopy` probably has some optimization built-in for such types.